### PR TITLE
Update README for Components directory

### PR DESCRIPTION
**Pull request by captainsafia**
Part of https://github.com/dotnet/aspnetcore/issues/27561.

### CODE REVIEW

File: src/Components/README.md [hunk 1]
```diff
@@ -1,17 +1,53 @@
...
 
-Blazor uses only the latest web standards. No plugins or transpilation needed. It runs in the browser on a real .NET runtime implemented in [WebAssembly](http://webassembly.org) that executes normal .NET assemblies.
+The following contains a description of each sub-directory in the `Components` directory.
 
-[![Gitter](https://badges.gitter.im/aspnet/blazor.svg)](https://gitter.im/aspnet/blazor?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
+- `Analyzers`: Contains a collection of Rosyln analyzers for Blazor components
+- `Authorization`: Contains source files associated with auth-related components and services in Blazor
+- `Components`: Contains the implementation for the Blazor component model
+- `Forms`: Contains source files for Form components in Blazor
+- `Ignitor`: Contains source files for inspecting Blazor apps
```
Comment (pranavkm) [inline src/Components/README.md:15]:
```suggestion
- `Ignitor`: A library for testing Blazor Server apps
```

File: src/Components/README.md [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+- `Samples`: Contains a collection of sample apps in Blazor
+- `Server`: Contains the implementation for Blazor Server-specific components
+- `Shared`: Contains a collection of shared constants and helper methods/classes
+- `Web`: Contains source files for handling DOM events, forms, and other components
+- `Web.JS`: Contains the source files for Blazor's client-side JavaScript
+- `WebAssembly`: Contains the implementation for WebAssembly-specific components
```
Comment (pranavkm) [inline src/Components/README.md:21]: Should we also document individual folders inside this directory? There's half a dozen of these.

File: src/Components/README.md [hunk 3]
```diff
@@ -1,17 +1,53 @@
...
 
-You can learn more about Blazor at https://blazor.net.
+## Development Setup
 
-## Getting Started
+### Build
+
+This project takes a dependency on MVC and SignalR packages. In order to build this project and its dependencies, run the following command inside this directory.
```
Comment (pranavkm) [inline src/Components/README.md:27]: What do you think about adding a general section to BuildFromSources.md that talks about how to build "sub-folders" and linking from here? 

> Each directory under src represents a feature area of ASP.NET Core. We provide convenience scripts `build.sh` or `build.cmd` inside these directories that let you build and test only this feature. For instance, if you were interested in contributing to Blazor, navigate to the `src/Components` directory, and run `build.[cmd|sh]` there to build all the projects and it's dependencies.

Then we link to these instructions from each of these docs. so we're not repeating ourselves?

Comment (pranavkm) [reply src/Components/README.md:27]: > This project takes a dependency on MVC and SignalR packages.

Ehh, these things are easy to go out of sync and don't really help the user much. Could we not list this here?

**Review by SteveSandersonMS — APPROVED** [review]
Looks good to me! I'll assume you and @pranavkm will figure out the details of accounting for his feedback.

File: src/Components/README.md (hunk 2 above)
Comment (captainsafia) [reply src/Components/README.md:21]: Yep. Added some content here.

File: src/Components/README.md (hunk 3 above)
Comment (captainsafia) [reply src/Components/README.md]: > Then we link to these instructions from each of these docs. so we're not repeating ourselves?

Good insight. I think we can probably address this for all the projects in one go as part of https://github.com/dotnet/aspnetcore/issues/27631.

> Ehh, these things are easy to go out of sync and don't really help the user much. Could we not list this here?

I've removed it.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**